app/models/remote_authenticatable.rb

```diff
@@ -94,6 +94,64 @@ def cognito_client_id
         Rails.application.secrets.cognito_client_id
       end
 
+      def update(params, *_options)
+        self.email = params[:email] if params.has_key?("email")
+        self.family_name = params[:family_name] if params.has_key?("family_name")
+        self.given_name = params[:given_name] if params.has_key?("given_name")
+        self.full_name = self.given_name + " " + self.family_name
+        self.organisation = params[:organisation] if params.has_key?("organisation")
+        self.roles = params[:roles] if params.has_key?("roles")
+        self.phone_number = params[:phone_number] if params.has_key?("phone_number")
+        self
+      end
+
+      def update_with_password(params, *options)
+        current_password = params.delete(:current_password)
+
+        if params[:password].blank?
+          params.delete(:password)
+          params.delete(:password_confirmation) if params[:password_confirmation].blank?
+        end
+
+        result = if valid_password?(current_password)
+                   update(params, *options)
+                 else
+                   assign_attributes(params, *options)
+                   valid?
+                   errors.add(:current_password, current_password.blank? ? :blank : :invalid)
+                   false
+                 end
```

Comment:
Does `update(params, *options)` return a boolean?

This might be better if it was pulled out to its' own method.